javatests/arcs/core/storage/driver/DatabaseDriverProviderTest.kt

```diff
@@ -122,6 +156,86 @@ class DatabaseDriverProviderTest {
     assertThat(singletonDriver.storageKey).isEqualTo(key)
   }
 
+  @Test
+  fun removeAllEntities() = runBlockingTest {
+    val managerMock = mock<DatabaseManager>()
+    provider.configure(managerMock, schemaHashLookup::get)
+
+    provider.removeAllEntities()
+
+    verify(managerMock).removeAllEntities()
+  }
+
+  @Test
+  fun removeEntitiesCreatedBetween() = runBlockingTest {
+    val managerMock = mock<DatabaseManager>()
+    provider.configure(managerMock, schemaHashLookup::get)
+
+    provider.removeEntitiesCreatedBetween(DUMMY_START, DUMMY_END)
+
+    verify(managerMock).removeEntitiesCreatedBetween(DUMMY_START, DUMMY_END)
+  }
+
+  @Test
+  fun getEntitiesCount_inMemory() = runBlockingTest {
+    val managerMock = mock<DatabaseManager>()
+    provider.configure(managerMock, schemaHashLookup::get)
+    whenever(managerMock.getEntitiesCount(false)).thenReturn(DUMMY_START)
+
+    val actual = provider.getEntitiesCount(inMemory=true)
+
+    verify(managerMock).getEntitiesCount(inMemory=false)
```

Comment:
this should be persistent=false (for some reason they are inverted), below as well